src/navigator/navigator.ts

```diff
@@ -122,9 +122,13 @@ export class Navigator {
       return false;
     }
 
-    return (
-      pos.spineItemIndex + 1 >= this.pub.spine.length && pos.pageIndex >= pos.spineItemPageCount
-    );
+    if (pos.view.fixedLayout() === true) {
+      return pos.spineItemIndex + 1 >= this.pub.spine.length;
+    } else {
+      return (
+        pos.spineItemIndex + 1 >= this.pub.spine.length && pos.pageIndex >= pos.spineItemPageCount
+      );
+    }
```

Comment:
I think the approach of the fix is good. Probably just refactor the code a bit to add a local variable to avoid code duplicate, like:
```
    const isLastSpineItem = pos.spineItemIndex + 1 >= this.pub.spine.length
    if (pos.view.fixedLayout() === true) {
      return isLastSpineItem;
    } else {
      return (
        isLastSpineItem && pos.pageIndex >= pos.spineItemPageCount
      );
    }